lib/cforum_web/templates/message/close_vote.html.eex

```diff
@@ -1,17 +1,23 @@
-<h3><%= gettext("close vote (%{votes}/%{votes_needed})", votes: CloseVotes.no_votes(@close_vote), votes_needed: ConfigManager.conf(@conn, "close_vote_votes")) %></h3>
-
-<p>
-  <%= OpenCloseVoteView.l10n_action(@conn, @close_vote) %>
-  <%= gettext("The given reason:") %>
-  <em><%= if @close_vote.reason == "custom", do: @close_vote.custom_reason, else: OpenCloseVoteView.l10n_reason(@close_vote.reason) %></em>
-</p>
-
-<%= if Abilities.may?(@conn, "messages/open_close_vote", :vote, %CloseVote{@close_vote | message: @message}) do %>
-  <p>
-    <%= if CloseVotes.has_voted?(@current_user, @close_vote) do %>
-      <%= VHelpers.Button.default_button(gettext("I changed my mind"), to: Path.oc_vote_path(@conn, @thread, @message, @close_vote), method: :patch) %>
-    <% else %>
-      <%= VHelpers.Button.default_button(gettext("I agree"), to: Path.oc_vote_path(@conn, @thread, @message, @close_vote), method: :patch) %>
-    <% end %>
-  </p>
-<% end %>
+<h3>
+  <%= if ConfigManager.conf(@conn, "close_vote_action_" <> @close_vote.reason) == "close" do %>
+    <%= gettext("close vote (%{votes}/%{votes_needed})", votes: CloseVotes.no_votes(@close_vote), votes_needed: ConfigManager.conf(@conn, "close_vote_votes")) %>
+  <% else %>
+    <%= gettext("delete vote (%{votes}/%{votes_needed})", votes: CloseVotes.no_votes(@close_vote), votes_needed: ConfigManager.conf(@conn, "close_vote_votes")) %>
+  <% end %>  
+</h3>
+
+<p>
+  <%= OpenCloseVoteView.l10n_action(@conn, @close_vote) %>
+  <%= gettext("The given reason:") %>

```

Comment:
hier bitte den Grund als Parameter in den Gettext-String, z.B. so: `gettext("The given reason: %{reason}", reason: vote_reason(@close_vote))`

Reply:
(ich weiß, dass ich das anders gemacht habe, aber das ist eine gute refactoring-möglichkeit)

Reply:
Mache ich gerne, dazu zwei Fragen:
- Was ist mit dem `<em>`? Bisher ist der Grund in `<em>` gesetzt, das ist vermutlich so nicht mehr möglich, wenn der Grund mit gettext geholt wird.
- Beim ersten drüberlesen sehe ich kein vote_reason, ich gehe davon aus, dass ich das anlegen muss? Dann würde ich unter lib/cforum_web/views/messages/open_close_vote_view.ex entsprechend eine neue Definition anlegen, die das abbildet, was da gerade im Moment noch im Template steht. Korrekt?

Reply:
Das `<em>` würde ich ersatzlos streichen. Bezüglich der Funktion: ja, die müsstest du erstellen, und ja, die OpenCloseVoteView wäre die richtige Stelle :)

Reply:
So langsam krieg ich den Dreh raus, glaub ich. Hoffe ich xD